integrations/ethereum.go

```diff
@@ -0,0 +1,92 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stellar/starbridge/model"
+)
+
+// FetchEthTxByHash returns a model.Transaction
+func FetchEthTxByHash(txHash string) (*model.Transaction, error) {
+	conn, e := ethclient.Dial("https://ropsten.infura.io/v3/a42ad89bbcec4ddca5c9abb60eb4a300")
+	if e != nil {
+		return nil, fmt.Errorf("could not connect to infura testnet: %s", e)
+	}
+
+	ctx := context.Background()
+	ethTxReceipt, e := conn.TransactionReceipt(ctx, common.HexToHash(txHash))
+	if e != nil {
+		return nil, fmt.Errorf("could not fetch transaction receipt '%s' by hash from network: %s", txHash, e)
+	}
+	ethTx, isPending, e := conn.TransactionByHash(ctx, common.HexToHash(txHash))
```

Comment:
we need both the tx receipt and pending tx information to get everything we want. I thought this would have been simpler for Ethereum but I'm surprised it wasn't.